### DISCUSSION TITLE
enabling uprobes for library and executable at the same time

### DISCUSSION THREAD

**Discussion by rafaelvanoni** [bpftrace/bpftrace#1692] [Q&A]
Hi folks, I was wondering if there's a way of enabling uprobes for shared library functions and the ones in an executable at the same time. Something like
```
# bpftrace -e 'uprobe::bar{...}' -c ./foo
```
I know I can use something like
```
 bpftrace -e 'uprobe:./foo:bar,uprobe:/lib/x86_64-linux-gnu/libc.so.6:baz{..}' -c ./foo
```
But it would make it simpler to write scripts, especially when using thread local variables to enable probes along a specific call stack.

Thanks

**Comment by fbs** [comment]
Sorry for the delay.

I'm not entirely sure what you mean. Do you want to be able to write `uprobe::bar,uprobe::baz` and have that attach both to the target binary as well as any `bar`s and `baz`s that show up in linked libraries?

**Reply by rafaelvanoni** [reply]
Sorry for the delay.

Yes, having to specify exactly where the function comes from is a bit clunky. Being allowed to run a more 'open ended' script would allow for a better debugging workflow imho.
